gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/impl/ApiKeyServiceImpl.java

```diff
@@ -657,6 +658,11 @@ private ApiKeyEntity convert(ExecutionContext executionContext, ApiKey apiKey) {
 
         apiKeyEntity.setDaysToExpirationOnLastNotification(apiKey.getDaysToExpirationOnLastNotification());
 
+        if (apiKey.getKey() != null) {
+            // Hash the API Key as used for authentication in the Kafka Gateway
+            apiKeyEntity.setHash(DigestUtils.md5DigestAsHex(apiKey.getKey().getBytes()));
+        }
```

Comment:
Maybe we could hash only if the API is Native ?

Reply:
There isn't a notion of `ApiType` with `ApiKeyEntity`, so it would not be very simple to return the hash only when the API is Native.